pkg/snapshot/restorer/restorer_test.go

```diff
@@ -0,0 +1,240 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restorer_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/pkg/types"
+	"github.com/gardener/etcd-backup-restore/pkg/miscellaneous"
+	"github.com/gardener/etcd-backup-restore/pkg/snapstore"
+
+	. "github.com/gardener/etcd-backup-restore/pkg/snapshot/restorer"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("Running Restorer", func() {
+	var (
+		logger *logrus.Logger
+
+		store snapstore.SnapStore
+		rstr  *Restorer
+
+		restoreCluster      string
+		restoreClusterToken string
+		restoreDataDir      string
+		restorePeerURLs     []string
+		restoreName         string
+		skipHashCheck       bool
+		maxFetchers         int
+
+		clusterUrlsMap types.URLsMap
+		peerUrls       types.URLs
+		baseSnapshot   *snapstore.Snapshot
+		deltaSnapList  snapstore.SnapList
+	)
+
+	BeforeEach(func() {
+		fmt.Println("Initializing snapstore and restorer")
+
+		logger = logrus.New()
+		restoreDataDir = etcdDir
+		restoreClusterToken = "etcd-cluster"
+		restoreName = "default"
+		restoreCluster = restoreName + "=http://localhost:2380"
+		restorePeerURLs = []string{"http://localhost:2380"}
+		clusterUrlsMap, err = types.NewURLsMap(restoreCluster)
+		Expect(err).ShouldNot(HaveOccurred())
+		peerUrls, err = types.NewURLs(restorePeerURLs)
+		Expect(err).ShouldNot(HaveOccurred())
+		skipHashCheck = false
+		maxFetchers = 6
+
+		err = corruptEtcdDir()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		store, err = snapstore.GetSnapstore(&snapstore.Config{Container: snapstoreDir, Provider: "Local"})
+		Expect(err).ShouldNot(HaveOccurred())
+
+		baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		rstr = NewRestorer(store, logger)
+	})
+
+	Context("with zero fetchers", func() {
+		fmt.Println("Testing for max-fetchers=0")
+		It("should return error", func() {
+			maxFetchers = 0
+
+			restoreOptions := RestoreOptions{
+				ClusterURLs:    clusterUrlsMap,
+				ClusterToken:   restoreClusterToken,
+				RestoreDataDir: restoreDataDir,
+				PeerURLs:       peerUrls,
+				SkipHashCheck:  skipHashCheck,
+				Name:           restoreName,
+				MaxFetchers:    maxFetchers,
+				BaseSnapshot:   *baseSnapshot,
+				DeltaSnapList:  deltaSnapList,
+			}
+			err = rstr.Restore(restoreOptions)
+			Expect(err).Should(HaveOccurred())
```

Comment:
Might be good to validate the error message also.

Reply:
Sure, will definitely look into it.